Assignment/Book-Management/src/app/core/services/book.service.ts

```diff
@@ -27,9 +35,87 @@ export class BookService {
         this.totalItems = objTemp.totalItems;
       })).subscribe()
 
-    return this.http.get<any>(environment.apiLink+`/books?pageNumber=${pageIndex}&pageSize=${pageSize}`, {
+    return this.http.get<any>(environment.apiLink+`/books?pageNumber=${pageIndex}&pageSize=${pageSize}`)
+  }
+
+  getBook(id: number) {
+    return this.http.get<any>(environment.apiLink+`/books/${id}`, {
+      headers: new HttpHeaders({
+        Authorization: 'Bearer ' + environment.token
+      })
+    })
+  }
+
+  getAuthor(id: number) {
+      this.author = {};
+      this.authorService.getAuthorById(id).subscribe(val => {
+        this.author = {
+          "id": val.id,
+          "name": val.name,
+          "website": val.website,
+          "birthday": `${val.birthday}`,
+          "cover": null
+        }
+      });
+  }
+
+  getCategories(ids: number[]) {
+    this.categories = [];
+    for (const id of ids) {
+        this.categoryService.getCategoryById(id).subscribe(val => {
+          this.categories.push({
+            "id": val.id,
+            "name": val.name,
+            "description": val.description
+          })
+        });
+      }
+  }
+
+  createBook(book: bookDb) {
+    let data = {
+      "name": book.bookName,
+      "description": book.description === "None" ? null : book.description,
+      "price": +book.price,
+      "year": +book.year,
+      "author": this.author,
+      "publisher": book.publisher,
+      "cover": null,
+      "categories": this.categories
+    }
+    return this.http.post(environment.apiLink+`/books/`, JSON.stringify(data) , {
+      headers: new HttpHeaders({
+          Authorization: 'Bearer ' + environment.token,
+          'Content-Type': 'application/json'
+      })
+    })
+  }
+
+  updateBook(id: number, book: bookDb ) {
+
+    let data = {
+      "name": book.bookName,
+      "description": book.description === "None" ? null : book.description,
+      "price": +book.price,
+      "year": +book.year,
+      "author": this.author,
+      "publisher": book.publisher,
+      "cover": null,
+      "categories": this.categories
+    }
```

Comment:
Same here.

Reply:
Done. Because I haven't merged yet, so it not update